moabb/datasets/liu2024.py

```diff
@@ -0,0 +1,310 @@
+        """Encode the columns 'value' and 'trial_type' into a single event type.
+
+        Parameters
+        ----------
+        events_df : pd.DataFrame
+            DataFrame containing the events information.
+
+        Returns
+        -------
+        np.ndarray
+            Array of encoded event types.
+
+        Notes
+        -----
+        The 'trial_type' variable can take the following values:
+         - 1 : Left hand
+         - 2 : Right hand
+
+        The 'value' variable can take the following values:
+         - 1 : instructions
+         - 2 : MI
+         - 3 : break
+
+        """
+        # Define the mapping dictionary
+        encoding_mapping = {
+            (2, 2): 1,  # Left hand, MI
+            (1, 2): 2,  # Right hand, MI
+            (1, 1): 3,  # Right hand, instructions
+            (1, 3): 4,  # Right hand, break
+            (2, 1): 3,  # Left hand, instructions
+            (2, 3): 4,  # Left hand, break
+        }
+
+        mapping = {
+            0: "left_hand",
+            1: "right_hand",
+            2: "instr",
+            3: "break",
+        }
+        # Apply the mapping to the DataFrame
+        event_category = events_df.apply(
+            lambda row: encoding_mapping[(row["trial_type"], row["value"])], axis=1
+        )
+
+        return event_category, mapping
+
+    def _get_single_subject_data(self, subject):
+        """Return the data of a single subject.
+
+        Parameters
+        ----------
+        subject : int
+            The subject number to fetch data for.
+
+        Returns
+        -------
+        dict
+            A dictionary containing the raw data for the subject.
+        """
+
+        file_path_list = self.data_path(subject)[0]
+        path_electrodes, path_events = self.data_infos()
+
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            # Read the subject's raw data
+            raw = mne.io.read_raw_edf(
+                file_path_list, verbose=False, infer_types=True, stim_channel=""
+            )
+        # Normalize and Read the montage
+        path_electrodes = self._normalize_extension(path_electrodes)
+        # Read and set the montage
+        montage = read_custom_montage(path_electrodes)
+
+        events_df = pd.read_csv(path_events, sep="\t")
+
+        # Encode the events
+        event_category, mapping = self.encoding(events_df=events_df)
+
+        events = self.create_event_array(raw=raw, event_category=event_category)
+
+        # Creating and setting annotations from the events
+        annotations = mne.annotations_from_events(
+            events, sfreq=raw.info["sfreq"], event_desc=mapping
+        )
+
+        raw = raw.set_annotations(annotations)
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            # Removing the stimulus channels
+            raw = raw.pick(["eeg", "eog"])
```

Comment:
I loaded the data of subject 1 and the channels HEOL and HEOR had an EEG type. These are EOG electrodes right?
Their type should be updated